### ISSUE TITLE
allow settings to be set in a settings file as well as on command line

### ISSUE THREAD

**Issue by cdepillabout** [arow-oss/goat-guardian#17]
Currently, goat-guardian only accepts settings from environment variables.

Eventually, it would be nice to allow settings to be set in a settings file (possibly YAML or dhall) as well as on the command line.

We will have to decide which settings override each other.  For example, if you specify both the `--twitter-oauth-key` command line flag and the `GG_TWITTER_OAUTH_KEY` environment variable, but with different values, which one takes precedence?

**Comment by arowM** [comment]
Basically, I do not agree to provide multiple ways to specify a same setting.
Each way (env vars, config files, command line args) has its pros. and cons.
(i.e., Some settings should be specified only by env vars, and other settings should be specified only by config files, ...)
It's important to use them properly for good end user UX.

For example, config files can treat arrays, so some complicated settings should be specified by config files, but they have defeats of hardness to inform their place to the daemon program.

Let's check how some reverse proxy apps do that.

**Comment by cdepillabout** [comment]
I agree that some settings will be too complicated to set via environment variables or command line flags.  For instance, the routing settings from #18 will probably only be able to be set in a config file, since they are so complicated.

However, for simple configuration options, like the `GG_TWITTER_OAUTH_KEY` example from above, I think it is best for operators to be able to specify options however they prefer, whether that is through environment variables, command line options, or a settings file.  This can make deployment and devops easier for operators when deploying a new service.

Whether or not we want to spend our limited time in supporting multiple styles of specifying options is a different question though!  The best course of action for us may be to wait for an enterprising developer on Github to send us a PR implementing this!

**Comment by arowM** [comment]
It makes sense.
> The best course of action for us may be to wait for an enterprising developer on Github to send us a PR implementing this!

Cool!
